modules/openy_gc_personal_training/src/PersonalTrainingSeriesManager.php

```diff
@@ -0,0 +1,369 @@
+   */
+  public function updateItemsOfSeries(int $series_id, array &$context): void {
+    $storage = $this->entityTypeManager->getStorage('personal_training');
+    $series = $storage->load($series_id);
+    if (!($series instanceof PersonalTrainingInterface)) {
+      return;
+    }
+    $values = $series->toArray();
+    $keys_to_exclude = [
+      'id',
+      'uuid',
+      'field_schedule',
+      'field_exclusions',
+      'type',
+      'date',
+    ];
+    $values = array_diff_key($values, array_flip($keys_to_exclude));
+
+    if (!isset($context['sandbox']['progress'])) {
+      $context['sandbox']['progress'] = 0;
+      $context['sandbox']['current'] = 0;
+      $context['sandbox']['max'] = $storage->getQuery()
+        ->condition('type', 'personal_training')
+        ->condition('field_parent', $series_id)
+        ->count()
+        ->execute();
+      $context['results']['op'] = $this->t('updated');
+    }
+
+    $ids = $storage->getQuery()
+      ->condition('type', 'personal_training')
+      ->condition('field_parent', $series_id)
+      ->condition('id', $context['sandbox']['current'], '>')
+      ->range(0, self::BATCH_LIMIT)
+      ->sort('id')
+      ->execute();
+    if (!$ids) {
+      return;
+    }
+    $items = $storage->loadMultiple($ids);
+    foreach ($items as $item) {
+      $context['sandbox']['current'] = $item->id();
+      foreach ($values as $key => $value) {
+        $item->set($key, $value);
+      }
+      $item->save();
+      $context['sandbox']['progress']++;
+      $context['message'] = $this->t('Updating PTs :progress of :count', [
+        ':progress' => $context['sandbox']['progress'],
+        ':count' => $context['sandbox']['max'],
+      ]);
+      $context['results']['processed'] = $context['sandbox']['progress'];
+    }
+
+    $context['finished'] = empty($context['sandbox']['max']) ? 1 : ($context['sandbox']['progress'] / $context['sandbox']['max']);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function deleteItemsOfSeries(int $series_id, array &$context): void {
+    $storage = $this->entityTypeManager->getStorage('personal_training');
+    if (!isset($context['sandbox']['progress'])) {
+      $context['sandbox']['progress'] = 0;
+      $context['sandbox']['current'] = 0;
+      $context['sandbox']['max'] = $storage->getQuery()
+        ->condition('type', 'personal_training')
+        ->condition('field_parent', $series_id)
+        ->count()
+        ->execute();
+      $context['results']['op'] = $this->t('deleted');
+    }
+
+    $ids = $storage->getQuery()
+      ->condition('type', 'personal_training')
+      ->condition('field_parent', $series_id)
+      ->condition('id', $context['sandbox']['current'], '>')
+      ->range(0, self::BATCH_LIMIT)
+      ->sort('id')
+      ->execute();
+    if (!$ids) {
+      return;
+    }
+    $items = $storage->loadMultiple($ids);
+    foreach ($items as $item) {
+      $context['sandbox']['current'] = $item->id();
+      $item->delete();
```

Comment:
What if the user adds PT to his favorites? Do we need to delete that reference?

Reply:
We have no choice there

Reply:
@NightHunterSV I think we need to react on deleting nodes/events/trainings and delete all referenced likes

cc @anpolimus

Reply:
Yes, sure. But in case of this task - only PT.
For nodes/events we need to raise an issue.